File: src/main/java/org/example/bankramenserver/infrastructure/integration/IntegrationOutboxScheduler.java
package org.example.bankramenserver.infrastructure.integration;

import lombok.RequiredArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class IntegrationOutboxScheduler {

    private final IntegrationOutboxDispatcher outboxDispatcher;

    @Scheduled(fixedDelayString = "${integration.dispatch-interval-ms:30000}")
    public void dispatchDue() {
        outboxDispatcher.dispatchDue();
    }
}

File: src/main/kotlin/org/example/balogserver/infrastructure/integration/IntegrationOutboxScheduler.kt
package org.example.balogserver.infrastructure.integration

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class IntegrationOutboxScheduler(
    private val outboxDispatcher: IntegrationOutboxDispatcher,
) {
    @Scheduled(fixedDelayString = "\${integration.dispatch-interval-ms:30000}")
    fun dispatchDue() = outboxDispatcher.dispatchDue()
}
